eng/tools/specs-model/eslint.config.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
```

Comment:
FYI it took me almost 3 hours to get this eslint work as intended, with no false positive typing errors, etc.

My take is to start with this very strict ruleset and gradually adjust the rules in the `rules: {` section, providing in comments our explanation why we adjusted some rules. This way we will gradually build explicit code conventions, immediately backed by automation.

Reply:
I'm fine using whatever rules you want as a starting point, and we can discuss specific points of disagreement.